### DISCUSSION TITLE
lazy.nvim not loading rustaceanvim

### DISCUSSION THREAD

**Discussion by rhaskia** [mrcjkb/rustaceanvim#108] [Q&A] [answered]
I've just set up rustaceanvim with lazy as shown in the readme, but it does not seem to load, and syntax highlighting takes a good few seconds to show up. I'm just wondering if I may have missed something that I need to set up for it to work.

**Comment by mrcjkb** [comment]
Hey 👋

rustaceanvim doesn't do any syntax highlighting - that would probably be nvim-treesitter.

This plugin doesn't load anything at startup - only when you open a rust file.
You can see if all the requirements are met by running `:checkhealth rustaceanvim`.

**Reply by rhaskia** [reply]
I tried that command and it says there is no healthcheck for rustaceanvim ):

**Reply by mrcjkb** [reply]
Then it's likely a config in your lazy.nvim setup preventing this plugin from being loaded (probably something to do with lazy loading).

You can follow [the troubleshooting guide](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#troubleshooting) and start from a minimal config, and then add your lazy.nvim settings to try and find out which one is causing it.

**Reply by rhaskia** [reply]
Could not get the minimal.lua to work, but I eventually found you can set lazy loading to false like this:
```
{
  'mrcjkb/rustaceanvim',
  version = '^3', -- Recommended
  ft = { 'rust' },
  lazy = false,
}
```

**Reply by mrcjkb** [reply]
🤔 That shouldn't necessary though.
Afaik, the `ft = { 'rust' },`  option tells lazy.nvim to load the plugin when a rust file is opened.

Either there's a bug in the lazy.nvim version you're using, or there's another setting somewhere disabling that (I don't use lazy.nvim myself so I'm afraid I can't really help with that).

Either way, lazy loading this plugin doesn't save you any startup time.

**Reply by rhaskia** [reply]
yeah, it's confusing me too, I only just installed lazy a few days ago just for this plugin. weirdly enough, even after forcing the plugin to activate syntax highlighting only shows properly on the main.rs file

**Reply by mrcjkb** [reply]
> syntax highlighting only shows properly on the main.rs file

That could be caused by an error, which prevented the syntax highlighting from being enabled.
You could try and reproduce it and run `:messages` to see if there are any error messages not being displayed.

**Reply by rhaskia** [reply, marked as answer]
Ah, managed to fix all the problems by removing the config I had in `~/.config/nvim/after/ftplugin/rust.lua`. I think something may have broken the file type loading from that. Thank you for all the help though!

**Reply by Mathijs-Bakker** [reply]
Wow! I spend a day on this!
I removed:
`~/.config/nvim/ftplugin/rust.lua.`
And it worked!
🙏🏼
